api/Api/Utilities/JsonListener.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Net;
+using System.Net.Sockets;
+using System.Reflection;
+using System.IO;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Schema;
+using Newtonsoft.Json.Schema.Generation;
+
+namespace SynthesisAPI.Utilities
+{
+    public static class JsonListener
```

Comment:
So a few things with this class:
1) The JsonListener should be standalone. Instead of a Listen function, there should be some kind of Start function that will create the server, spawn a thread, and continuously listen. Then make sure you have the ability to forcibly interrupt the server whenever.
2) If the IDs are persistent and you are getting data more that once, using `Packets.Add()` is going to throw exceptions because it assumes that key doesn't exist. Use `Packets[key] = value`. This will work whether or not a kvp exists already.
3) If the packets you are receiving are longer than 256 bytes (which because its json it likely will be) currently your parser cuts the packet off at 256 bytes. Consider expanding the maximum sized byte buffer or add some logic to find the end of a given packet (I think you can look for a line termination character if you serialize the json correctly. If you use this approach I'd suggest just making your life easier and using a `StreamReader` which has a `ReadLine()` method just for this). This could also try to parse 2 packets written since the last read as a single packet.
4) Not sure why you are re-sending the packet back to the client. If the client tries to deserialize it, the client will likely have issues with all the variable names being altered by doing `.ToUpper()`